tests/std/tests/P0475R1_P0591R4_uses_allocator_construction/test.cpp

```diff
@@ -179,6 +179,17 @@ void test_GH_2021() { // COMPILE-ONLY
     tags[0];
 }
 
+struct MoveOnlyType {
+    MoveOnlyType()               = default;
+    MoveOnlyType(MoveOnlyType&&) = default;
+};
+
+void test_LWG3527() { // COMPILE-ONLY
+    std::allocator<MoveOnlyType> alloc;
+    auto p                  = pair<MoveOnlyType&&, MoveOnlyType&&>{MoveOnlyType{}, MoveOnlyType{}};

```

Comment:
Although this is compile-only, this creates rvalue references that are immediately dangling. It would be less confusing (and less likely to emit warnings in the future) to bind these references to a persistent object.